itest/psbt_test.go

```diff
@@ -1610,6 +1618,352 @@ func testPsbtSighashNoneInvalid(t *harnessTest) {
+	require.Len(t.t, vPkt.Outputs, 1)
+
+	// On the vPSBT level, which describes the assets transfer, we do not
+	// commit to any outputs.
+	vPkt.Inputs[0].SighashType = txscript.SigHashNone
+
+	// Let's do some sanity checks on the structure of the vPSBT, and
+	// prepare its outputs before signing.
+	require.Equal(t.t, vPkt.Outputs[0].Type, tappsbt.TypeSimple)
+	require.NoError(t.t, tapsend.PrepareOutputAssets(ctxb, vPkt))
+	require.Nil(t.t, vPkt.Outputs[0].Asset.SplitCommitmentRoot)
+	require.Len(t.t, vPkt.Outputs[0].Asset.PrevWitnesses, 1)
+	require.Nil(t.t, vPkt.Outputs[0].Asset.PrevWitnesses[0].SplitCommitment)
+
+	fundedPsbtBytes, err := tappsbt.Encode(vPkt)
+	require.NoError(t.t, err)
+
+	// Alice signs the vPSBT.
+	signedResp, err := alice.SignVirtualPsbt(
+		ctxb, &wrpc.SignVirtualPsbtRequest{
+			FundedPsbt: fundedPsbtBytes,
+		},
+	)
+	require.NoError(t.t, err)
+	require.Contains(t.t, signedResp.SignedInputs, uint32(0))
+
+	// Deserialize the signed vPSBT.
+	vPkt, err = tappsbt.Decode(signedResp.SignedPsbt)
+	require.NoError(t.t, err)
+
+	// Now we need to create the bitcoin PSBT where the previously created
+	// vPSBT will be anchored to.
+	btcpsbt, err := tapsend.PrepareAnchoringTemplate([]*tappsbt.VPacket{
+		vPkt,
+	})
+	require.NoError(t.t, err)
+
+	// This bitcoin PSBT should have 1 input, which is the anchor of Alice's
+	// assets, and 2 outputs that correspond to Alice's bitcoin change
+	// (index 0) and the anchor that carries the assets (index 1).
+	require.Len(t.t, btcpsbt.Inputs, 1)
+	require.Len(t.t, btcpsbt.Outputs, 2)
+
+	// Let's set an actual address for Alice's output.
+	addrResp := t.lndHarness.Alice.RPC.NewAddress(&lnrpc.NewAddressRequest{
+		Type: lnrpc.AddressType_TAPROOT_PUBKEY,
+	})
+
+	aliceP2TR, err := btcutil.DecodeAddress(
+		addrResp.Address, harnessNetParams,
+	)
+	require.NoError(t.t, err)
+
+	alicePkScript, err := txscript.PayToAddrScript(aliceP2TR)
+	require.NoError(t.t, err)
+
+	// These are basically Alice's terms that she signs the assets over:
+	// Send me 69420 satoshis to this address that belongs to me, and you
+	// will get assets in return.
+	btcpsbt.UnsignedTx.TxOut[0].PkScript = alicePkScript
+	btcpsbt.UnsignedTx.TxOut[0].Value = 69420
+	derivation, trDerivation := getAddressBip32Derivation(
+		t.t, addrResp.Address, t.lndHarness.Alice,
+	)
+
+	// Add the derivation info and internal key for alice's taproot address.
+	btcpsbt.Outputs[0].Bip32Derivation = []*psbt.Bip32Derivation{
+		derivation,
+	}
+	btcpsbt.Outputs[0].TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{
+		trDerivation,
+	}
+	btcpsbt.Outputs[0].TaprootInternalKey = trDerivation.XOnlyPubKey
+
+	var b bytes.Buffer
+	err = btcpsbt.Serialize(&b)
+	require.NoError(t.t, err)
+
+	// Now we need to commit the vPSBT and PSBT, creating all the related
+	// proofs for this transfer to be valid.
+	resp, err := alice.CommitVirtualPsbts(
+		ctxb, &wrpc.CommitVirtualPsbtsRequest{
+			VirtualPsbts: [][]byte{signedResp.SignedPsbt},
```

Comment:
Hmm, not sure if I'm following this properly, but it appears that we're anchoring the vPSBT into the transaction above that has 1-input-1-output. Instead, Bob should be the one anchoring the vPSBT (with the blank signed check from Alice) into an output that _he_ solely controls.

Reply:
Actually scratch the above, I see an output index of `1` is provided. However how can we anchor this ourselves if we don't yet know Bob's public key? In other words, Bob should be the one that takes the vPSBT, and anchors it. Alice's role is to give the signed PSBT that spends her input with an asset, and asks for X amt of BTC by providing less total inputs than the output amt.

Reply:
Ok scratch all that, I see this was just a round about way of doing this, and we'll strip things down to just 1 input 1 output. 

Leaving these comments in place just to demonstrate my thought process while reviewing this.